src/dune_rules/rule_mode_decoder.mli

```diff
@@ -15,3 +15,4 @@ end
 
 val decode : Rule.Mode.t Dune_lang.Decoder.t
 val field : Rule.Mode.t Dune_lang.Decoder.fields_parser
+val is_ignored : Rule.Mode.t -> until_clean:[ `Ignore | `Keep ] -> bool
```

Comment:
I think it would be useful to have a high-level doc-string to explain what the setting of `until_clean` means.

Reply:
Done.